### PR TITLE
integrity(ticdc): add integrity package and move some files, reorder columns at the mounter

### LINKED ISSUE #8916: checksum mistmatch since the column is not ordered by the column id at the mounter
### What did you do?

```
drop database if exists test;
create database test;
use test;

create table tp_int
(
    id          int auto_increment,
    c_tinyint   tinyint   null,
    c_smallint  smallint  null,
    c_mediumint mediumint null,
    c_int       int       null,
    c_bigint    bigint    null,
    constraint pk
        primary key (id)
);

insert into tp_int(c_tinyint, c_smallint, c_mediumint, c_int, c_bigint) values (1, 2, 3, 4, 5);

alter table tp_int add column a int not null;
insert into tp_int(c_tinyint, c_smallint, c_mediumint, c_int, c_bigint, a) values (2, 3, 4, 5, 6, 7);

alter table tp_int modify c_tinyint bigint not null;
insert into tp_int(c_tinyint, c_smallint, c_mediumint, c_int, c_bigint, a) values (3, 4, 5, 6, 7, 8);

alter table tp_int drop column c_smallint;
insert into tp_int(c_tinyint, c_mediumint, c_int, c_bigint, a) values (4, 5, 6, 7, 8);

-- this should trigger backfill
alter table tp_int modify c_mediumint varchar(10) not null;
insert into tp_int(c_tinyint, c_mediumint, c_int, c_bigint, a) values (5, "234", 6, 7, 8);
```

### What did you expect to see?

the checksum matched

### What did you see instead?

checksum not matced, since the column is correctly ordered

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
(paste TiCDC version here)
```

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #8916 

### What is changed and how it works?

* Add `integrity` package, and move all related code to it.
* If the checksum enabled, reorder columns by the column id to get the correct checksum value.
* Add a new struct `Checksum`, and set it as a field of the row changed event to reduce memory allocation.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: cdc/entry/mounter.go [hunk 1]
```diff
@@ -446,6 +446,8 @@ func (m *mounter) verifyChecksum(
 
 	// the first checksum matched, it hits in the most case.
 	if checksum == first {
+		log.Info("checksum matched",
```
Comment (nongfushanquan) [inline cdc/entry/mounter.go:449]: too many logs will appear here

Comment (3AceShowHand) [reply cdc/entry/mounter.go:449]: I will change this to the debug level.

**Review by nongfushanquan — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**